pkg/model/core/stack.go

```diff
@@ -14,6 +18,9 @@ type Stack interface {
 	// Add a resource into stack.
 	AddResource(res Resource) error
 
+	// Get a resource by its id and type
+	GetResource(id string, resType Resource) (Resource, error)
+
```

Comment:
I agree it would probably simplify things to simply remove the modeling step entirely. Something else I noticed was the code has to be more complicated because it handles both delete and upsert scenarios. Eliminating the model may allow us to further simplify the delete logic as well.

Reply:
Make sense to remove generic stack, created a issue for it: https://github.com/aws/aws-application-networking-k8s/issues/463